.github/bin/runtests.sh

```diff
@@ -0,0 +1,157 @@
+#    BE LIABLE FOR PERSONAL INJURY, OR ANY INCIDENTAL, SPECIAL, INDIRECT OR
+#    CONSEQUENTIAL DAMAGES WHATSOEVER, INCLUDING, WITHOUT LIMITATION, DAMAGES
+#    FOR LOSS OF PROFITS, CORRUPTION OR LOSS OF DATA, FAILURE TO TRANSMIT OR
+#    RECEIVE ANY DATA OR INFORMATION, BUSINESS INTERRUPTION OR ANY OTHER
+#    COMMERCIAL DAMAGES OR LOSSES, ARISING OUT OF OR RELATED TO YOUR USE OR
+#    INABILITY TO USE THIS SHELL SCRIPT OR SERVICES OR ANY THIRD PARTY
+#    SOFTWARE OR APPLICATIONS IN CONJUNCTION WITH THIS SHELL SCRIPT OR
+#    SERVICES, HOWEVER CAUSED, REGARDLESS OF THE THEORY OF LIABILITY (CONTRACT,
+#    TORT OR OTHERWISE) AND EVEN IF THE AUTHOR HAS BEEN ADVISED OF THE
+#    POSSIBILITY OF SUCH DAMAGES. SOME JURISDICTIONS DO NOT ALLOW THE EXCLUSION
+#    OR LIMITATION OF LIABILITY FOR PERSONAL INJURY, OR OF INCIDENTAL OR
+#    CONSEQUENTIAL DAMAGES, SO THIS LIMITATION MAY NOT APPLY TO YOU. In no event
+#    shall THE AUTHOR's total liability to you for all damages (other than as may
+#    be required by applicable law in cases involving personal injury) exceed
+#    the amount of five dollars ($5.00). The foregoing limitations will apply
+#    even if the above stated remedy fails of its essential purpose.
+################################################################################
+
+# bootstrap checks (pre pre-checks)
+EXIT_CODE=0
+
+test -x `command -v uname` || exit 1 ;
+test -x `command -v git` || exit 1 ;
+
+if [[ ( $(uname -s) = "Darwin" ) ]] ; then
+	export CI_OS_NAME="osx" ;
+else
+	export CI_OS_NAME="linux" ;
+fi
+
+
+# if locking was needed:
+####
+# ulimit -t 600
+# PATH="/bin:/sbin:/usr/sbin:/usr/bin"
+# umask 137
+#
+LOCK_FILE="/tmp/CI_test_script_lock"
+ 
+if [[ -f ${LOCK_FILE} ]] ; then
+	echo "Lock found! ABORT"
+ 	exit ${EXIT_CODE:-0} ;
+fi
+# 
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGABRT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGHUP
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGTERM
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGQUIT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGINT
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit 1 ;' SIGKILL
+trap 'rm -f ${LOCK_FILE} 2>/dev/null || true ; wait ; exit ${EXIT_CODE:-0} ;' EXIT
+# 
+touch ${LOCK_FILE} 2>/dev/null || exit 0 ;
+
+# THIS IS THE ACTUAL WORK:
+
+export CI_TOOLS_BIN=$(dirname "${0}") ;
+#TODO: make this more absolute
+export CI_TEST_ROOT=$(dirname ${PWD:-./}/.) ;
+
+${CI_TOOLS_BIN}/pretests.sh
+
+function fn_test_pass() {
+	echo "... Passed" ;
+	return 0 ;
+}
+
+function fn_test_fail() {
+	echo "... FAILED" ;
+	return 1 ;
+} 1>&2
+
+function fn_test_cmd() {
+	local CI_TEST_CASE_RESULT=0 ;
+	local OLDUMASK=$(umask) ;
+	( ${CI_TOOLS_BIN%/}/${1} ${@:2:$#-1} ) || CI_TEST_CASE_RESULT=$? ;
+	wait ;
+	# clean up
+	umask "$OLDUMASK" ;
+	# return to start path
+	cd "${CI_TEST_ROOT}" ;
+	return $CI_TEST_CASE_RESULT ;
+}
```

Comment:
future tests can be run this function calling those future scripts in a hopefully maintainable way.